app/MetadataResolver/IwaraResolver.php

```diff
@@ -19,40 +18,62 @@ public function __construct(Client $client)
 
     public function resolve(string $url): Metadata
     {
-        $res = $this->client->get($url);
-        $metadata = new Metadata();
-        $html = (string) $res->getBody();
-        $crawler = new Crawler($html);
-
-        $infoElements = $crawler->filter('#video-player + div, .field-name-field-video-url + div, .field-name-field-images + div');
-        $title = $infoElements->filter('h1.title')->text();
-        $author = $infoElements->filter('.username')->text();
-        $description = $infoElements->filter('.field-type-text-with-summary')->text('');
-        $tags =  $infoElements->filter('a[href^="/videos"], a[href^="/images"]')->extract('_text');
-        // 役に立たないタグを削除する
-        $tags = array_values(array_diff($tags, ['Uncategorized', 'Other']));
-        array_push($tags, $author);
-
-        $metadata->title = $title;
-        $metadata->description = '投稿者: ' . $author . PHP_EOL . trim($description);
-        $metadata->tags = $tags;
-
-        // iwara video
-        if ($crawler->filter('#video-player')->count()) {
-            $metadata->image = 'https:' . $crawler->filter('#video-player')->attr('poster');
-        }
+        preg_match('~iwara\.tv/(?P<type>video|image)[s]?/(?P<id>\w+)~', $url, $m);
 
-        // youtube
-        if ($crawler->filter('iframe[src^="//www.youtube.com"]')->count()) {
-            if (preg_match('~youtube\.com/embed/(\S+)\?~', $crawler->filter('iframe[src^="//www.youtube.com"]')->attr('src'), $matches) === 1) {
-                $youtubeId = $matches[1];
-                $metadata->image = 'https://img.youtube.com/vi/' . $youtubeId . '/maxresdefault.jpg';
-            }
+        if (!isset($m['type']) || !isset($m['id'])) {
+            throw new \RuntimeException("Invalid URL: $url");
         }
 
-        // images
-        if ($crawler->filter('.field-name-field-images')->count()) {
-            $metadata->image = 'https:' . $crawler->filter('.field-name-field-images a')->first()->attr('href');
+        $metadata = new Metadata();
+
+        $apiUrl = "https://api.iwara.tv/{$m['type']}/{$m['id']}";
+        $res = $this->client->get($apiUrl);
+        $json = json_decode($res->getBody(), true);
+
+        $metadata->title = $json['title'];
+        $metadata->description = "投稿者: {$json['user']['name']}" . PHP_EOL . trim(str_replace("\n\n", "\n", $json['body']));
+
+        $excludeTags = ['uncategorized', 'other'];
+        // excludeTagsのidを持つタグを除外し、配列のindexを0から再構成する
+        $metadata->tags = array_values(array_diff(array_column($json['tags'], 'id'), $excludeTags));
+        // 投稿者のusernameをタグに追加する
+        // nameにはスペースなどの空白が含まれていることがあるためusernameを使用する
+        array_push($metadata->tags, $json['user']['username']);
```

Comment:
`$json['user']['username']` はstringですか? できれば、配列への要素追加は構文のほうを使うようにしたいです。  
(取るに足らない) 性能上の理由と、リポジトリ内を見渡した上での習慣的な理由です。

```suggestion
        $metadata->tags[] = $json['user']['username'];
```